scripts/irods/upgrade_configuration.py

```diff
@@ -345,6 +345,21 @@ def update_base(base, updates):
     if "irods_tcp_keepalive_time_in_seconds" in service_account_environment:
         new_server_config["tcp_keepalive_time_in_seconds"] = service_account_environment.get("irods_tcp_keepalive_time_in_seconds", None)
 
+    # Update database plugin config (if present) to new format
+    if len(new_server_config['plugin_configuration']['database']) != 0:
+        # Only focus on part we care about
+        database_config_options = new_server_config['plugin_configuration']['database']
+
+        # Assume we have a previously valid config (i.e. one database config property)
+        db_name = database_config_options.keys()[0]
+
+        # Transfer over the previous items (assuming property starts with 'db_')
+        for key in database_config_options[db_name].keys():
+            # First check that the prefix exists
+            if key[:3] == 'db_':
+                # Create new configuration item and remove old config option
+                database_config_options[key[3:]] = database_config_options.pop(key)
```

Comment:
We should forward unknown properties into the config in case the admin cares/needs them for other reasons.

Reply:
This should still forward the unknown properties. What it is doing is truncating the "db_*" properties, an example is the following:

```diff
{
-   "db_test": "Hello",
+   "test": "Hello",
    "misc": "I remain unmodified!"
}
```

Reply:
Ah, I see now.

The unknown properties are already there. This code just updates the keys.

Reply:
Please resolve this comment if there's nothing actionable here.